### PR TITLE
(PC-33579) feat(categoryButtonsSearch): new Design

### PR DESCRIPTION
**Pull request by nmajorfrances-pass**
Link to JIRA ticket: https://passculture.atlassian.net/browse/PC-33579

## Flakiness

If I had to re-run tests in the CI due to flakiness, I add the incident [on Notion][2]

## Checklist

I have:

- [x] Made sure my feature is working on web.
- [x] Made sure my feature is working on mobile (depending on relevance : real or virtual devices)
- [x] Written **unit tests** native (and web when implementation is different) for my feature.
- [x] Added a **screenshot** for UI tickets or deleted the screenshot section if no UI change
- [ ] If my PR is a bugfix, I add the link of the "résolution de problème sur le bug" [on Notion][1]
- [x] I am aware of all the best practices and respected them.

## Screenshots

**delete** _if no UI change_

| Platform         | Mockup/Before | After |
| :--------------- | :-----------: | :---: |
|  iOS    |Capture d’écran 2024-12-20 à 10 49 30|Simulator Screenshot - iPhone 15 Pro - 2024-12-20 at 11 03 52|
| Desktop - Chrome | Capture d’écran 2024-12-20 à 10 23 44|Capture d’écran 2024-12-20 à 11 03 46|

[1]: https://www.notion.so/passcultureapp/R-solution-de-probl-mes-sur-les-bugs-5dd6df8f6a754e6887066cf613467d0a
[2]: https://www.notion.so/passcultureapp/cb45383351b44723a6f2d9e1481ad6bb?v=10fe47258701423985aa7d25bb04cfee&pvs=4

## Best Practices

<details>
  <summary>Click to expand</summary>
These rules apply to files that you make changes to.
If you can't respect one of these rules, be sure to explain why with a comment.
If you consider correcting the issue is too time consuming/complex: create a ticket. Link the ticket in the code.

- In the production code: remove type assertions with `as` (type assertions are removed at compile-time, there is no runtime checking associated with a type assertion. There won’t be an exception or `null` generated if the type assertion is wrong). In certain cases `as const` is acceptable (for example when defining readonly arrays/objects). Using `as` in tests is tolerable.
- Remove bypass type checking with `any` (when you want to accept anything because you will be blindly passing it through without interacting with it, you can use `unknown`). Using `any` in tests is tolerable.
- Remove non-null assertion operators (just like other type assertions, this doesn’t change the runtime behavior of your code, so it’s important to only use `!` when you know that the value can’t be `null` or `undefined`).
- Remove all `@ts-expect-error` and `@eslint-disable`.
- Remove all warnings, and errors that we are used to ignore (`yarn test:lint`, `yarn test:types`, `yarn start:web`...).
- Use `gap` (`ViewGap`) instead of `<Spacer.Column />`, `<Spacer.Row />` or `<Spacer.Flex />`.
- Don't add new "alias hooks" (hooks created to group other hooks together). When adding new logic, this hook will progressively become more complex and harder to maintain.
- Remove logic from components that should be dumb.

Test specific:

- Avoid mocking internal parts of our code. Ideally, mock only external calls.
- When you see a local variable that is over-written in every test, mock it.
- Prefer `user` to `fireEvent`.
- When mocking feature flags, use `setFeatureFlags`. If not possible, mention which one(s) you want to mock in a comment (example: `jest.spyOn(useFeatureFlagAPI, 'useFeatureFlag').mockReturnValue(true) // WIP_NEW_OFFER_TILE in renderPassPlaylist.tsx` )
- In component tests, replace `await act(async () => {})` and `await waitFor(/* ... */)` by `await screen.findBySomething()`.
- In hooks tests, use `act` by default and `waitFor` as a last resort.
- Make a snapshot test for pages and modals ONLY.
- Make a web specific snapshot when your web page/modal is specific to the web.
- Make an a11y test for web pages.

Advice:

- Use TDD
- Use Storybook
- Use pair programming/mobs
</details>

### CODE REVIEW

File: src/features/home/pages/ThematicHome.tsx [hunk 1]
```diff
@@ -128,15 +128,17 @@ export const ThematicHome: FunctionComponent = () => {
...
-        homeEntryId: id,
-        from: params.from,
-        moduleId: params.moduleId,
-        moduleListId: params.moduleListId,
-      })
+    if (!id) {
+      return
     }
-  }, [id, params.from, params.moduleId, params.moduleListId])
+    analytics.logConsultThematicHome({
```
Comment (nmajorfrances-pass) [inline src/features/home/pages/ThematicHome.tsx:134]: @bebstein-pass j'ai pas trouvé comment géré cet analytics sans casser tout le typage

File: src/features/home/pages/ThematicHome.native.test.tsx [hunk 2]
```diff
@@ -172,15 +172,17 @@ describe('ThematicHome', () => {
...
   describe('analytics', () => {
-    it('should log ConsultHome', async () => {
+    it('should log ConsultThematicHome', async () => {
       renderThematicHome()
 
-      await waitFor(() => {
-        expect(analytics.logConsultHome).toHaveBeenNthCalledWith(1, { homeEntryId: 'fakeEntryId' })
+      await screen.findByText('Suivre')
+
+      expect(analytics.logConsultThematicHome).toHaveBeenNthCalledWith(1, {
```
Comment (bebstein-pass) [inline src/features/home/pages/ThematicHome.native.test.tsx:180]: on a juste fait une nouvelle fonction qui fait exactement comme l'ancienne

c'est censé etre un refactoring invisible au niveau des tests

c'est intéressant que les tests aient échoués

ça montre qu'on mock un niveau trop haut et qu'on test un détail d'implémentation

File: src/features/search/components/DumbCategoriesList/DumbCategoriesList.tsx [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  hideVenueMapLocationModal: () => void
+  children?: never
+}
+
+const DESKTOP_CATEGORY_BUTTON_HEIGHT = getSpacing(42)
+const MOBILE_CATEGORY_BUTTON_HEIGHT = getSpacing(36)
+const MOBILE_GAPS_AND_PADDINGS = getSpacing(2)
+const DESKTOP_GAPS_AND_PADDINGS = getSpacing(4)
+
+export const DumbCategoriesList: FunctionComponent<Props> = ({
```
Comment (bebstein-pass) [inline src/features/search/components/DumbCategoriesList/DumbCategoriesList.tsx:43]: personnellement, je préfèrerais `CategoriesListDumb` car j'aurais plutot tendance à taper `Cat` et laisser l'autocompletion trouver plutot que de taper `Dumb`

File: src/ui/theme/newColorMapping.ts [hunk 4]
```diff
@@ -0,0 +1,39 @@
+import { Color } from 'features/home/types'
+// eslint-disable-next-line no-restricted-imports
+import { ColorsEnum } from 'ui/theme/colors'
+
+export const newColorMapping: Record<
```
Comment (tprache-pass) [inline src/ui/theme/newColorMapping.ts:5]: On ne veut pas l'appeler juste ColorMapping ?

**Review by tprache-pass — APPROVED** [review]
Trop bien, ca nettoie en plus j'adore ✨

File: src/ui/theme/newColorMapping.ts (hunk 4 above)
Comment (nmajorfrances-pass) [reply src/ui/theme/newColorMapping.ts:5]: J'ai hésité, je me demandais si il fallait attendre qu'on ai officiellement plus les anciennes

**Pull request merged**